Firestore/core/src/firebase/firestore/model/field_value.h

```diff
@@ -216,27 +179,75 @@ class FieldValue {
     std::unique_ptr<ReferenceValue> reference_value_;
     std::unique_ptr<GeoPoint> geo_point_value_;
     std::unique_ptr<std::vector<FieldValue>> array_value_;
-    std::unique_ptr<ObjectValue> object_value_;
+    std::unique_ptr<Map> object_value_;
   };
 };
 
-// TODO(rsgowman): Expand this to roughly match the java class
-// c.g.f.f.model.value.ObjectValue. Probably move it to a similar namespace as
-// well. (FieldValue itself is also in the value package in java.) Also do the
-// same with the other FooValue values that FieldValue can return.
-struct ObjectValue {
-  // TODO(rsgowman): These will eventually be private. We do want the serializer
-  // to be able to directly access these (possibly implying 'friend' usage, or a
-  // getInternalValue() like java has.)
-  using Map = immutable::SortedMap<std::string, FieldValue>;
-  Map internal_value;
+/** A structured object value stored in Firestore. */
+class ObjectValue {
+ public:
+  ObjectValue(FieldValue fv) : fv_(std::move(fv)) {
```

Comment:
From our offline discussion, this should be `explicit`.

Reply:
Done.